### ISSUE TITLE
Finalize initializer syntax

### ISSUE THREAD

**Issue by bradcray** [chapel-lang/chapel#8777]
As a Chapel Programmer / Chapel Implementer, I would like the syntactic choices around initializers to be finalized so that I can start coding to the new initializer design without having to change code later if/when the names change.

Acceptance criteria:
Syntax around initializers is ratified by the team and documented.

**Comment by cassella** [comment]
done?

**Comment by bradcray** [comment]
I don't think so... we're still wrestling with the relationship between initializers and partial instantiation (e.g., issue #12318).
